File: python/number_guess.py
import random


print("Hi! Welcome to the Number Guessing Game\nYou have 7 chances to guess the correct number. Let's start!")

#Takes in input from the user and transforms them into Integers.
low = int(input("Enter the lower bound of the range: "))
high = int(input("Enter the upper bound of the range: "))

#Checks if the Lower bound is higher or equal to the Upper bound
while low >= high:
    print("Your lower bound is equal or higher than the upper bound. Please change your lower bound.")
    low = int(input("Enter the lower bound of the range: "))
    high = int(input("Enter the upper bound of the range: "))
    
print(f"\nYou have 7 chances to guess a number between {low} and {high}. Let's get started!")


random_number = random.randint(low, high)
chances = 7
#Limits the amount of chances for the user. Can change later to scale with range size.
guess_counter = 0

#Main loop logic. Will loop until the guess counter is equal to chances or if the right number was found.
while guess_counter < chances:
    #Check to see if User input was a number and inside the range of the bounds.
    while True:
        try:
            user_guess = int(input("Enter your guess: "))
            while user_guess > high or user_guess < low:
                print("Your guess is out of bounds.")
                user_guess = int(input("Enter your guess: "))
            break
        except ValueError:
            print("That is not a number. Try again!")  
            
    guess_counter += 1
    
    if (user_guess < random_number):
            print(f"You are too low!\nYou have {chances-guess_counter} chances left!")
    elif (user_guess > random_number):
            print(f"You are too high!\nYou have {chances-guess_counter} chances left!")
    else:
            print(f"Congratulations! You guessed the correct number in {guess_counter} guesses!")
            break
        
if (guess_counter == chances):
        print(f"Oh no! You have used all of your {chances} chances. Too bad!")

File: c/src/number_guess.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    printf("Hi! Welcome to the Number Guessing Game\nYou have 7 chances to guess the correct number. Let's start! \n");
    // Create an integer variable that will store the number we get from the user
    int low, high, random_number, guesses;
    int chances = 7;

    // Ask the user to type a number
    printf("Enter your lower bound: \n");
    scanf("%d", &low);

    printf("Enter your higher bound: \n");
    scanf("%d", &high);
    
    srand(time(0));
    random_number = (rand() % (high - low +1)) +low;

    printf("You have %d chances to guess a number between %d and %d. Let's get started!", chances,low,high); 

    while (guesses < chances) {
        guesses++;
    }
    return 0;
}
